src/playercards/customizable/SummonedServitorUpgradeSheet.ttslua

```diff
@@ -31,27 +31,54 @@ existingBoxes = { 1, 1, 1, 1, 1, 2, 3, 5 }
 
 inputBoxes = {}
 
+-- Locations of the skill selectors
+local SLOT_ICON_POSITIONS = {
+    arcane = { x = 0.160, z = 0.65},
+    ally = { x = -0.073, z = 0.65}
+  }
+  
+  local selectedSlots = {
+    arcane = false,
+    ally = false
+  }
+
 -- override 'marked boxes' for debugging ('all' or 'none')
 markDEBUG = ""
 
 -- save state when going into bags / decks
 function onDestroy() self.script_state = onSave() end
 
-function onSave() return JSON.encode({ markedBoxes, inputValues }) end
+function onSave()
+    local slotArray = { }
+    for slot, isSelected in pairs(selectedSlots) do
+      if (isSelected) then
+        table.insert(slotArray, slot)
+      end
+    end
+    return JSON.encode({ markedBoxes, inputValues, slotArray })
+  end
 
 -- Startup procedure
 function onLoad(saved_data)
     if saved_data ~= "" and markDEBUG == "" then
         local loaded_data = JSON.decode(saved_data)
         markedBoxes = loaded_data[1]
         inputValues = loaded_data[2]
+        log(loaded_data[3])
+        for _, slot in ipairs(loaded_data[3]) do
```

Comment:
This throws an error when loaded with an old save state.  Add a nil check to make it more resilient

Reply:
Additionally, this PR should update the script state accordingly

Reply:
Not sure what might be going on there, but as long as the load handles it if it's missing it shouldn't need to be updated.  Would be nice, but not critical.

Reply:
> For some reason whenever I try to stage the changes to the script state, it doesn't resolve? I get a warning about line endings and then git tells me there were actually no changes to the script state to begin with (i.e. nothing to stage anymore).

Just to be sure, are you changing this line? https://github.com/argonui/SCED/blob/main/objects/AllPlayerCards.15bb07/LivingLinkUpgradeSheet.19a05b.json#L37

Reply:
I got it now!  I assumed the mod manager would be updating these automatically, that was my mistake.  Went in and manually updated, should work correctly now.